docs/pages/pmd/userdocs/extending/writing_java_rules.md

```diff
@@ -13,9 +13,11 @@ author: Tom Copeland <tomcopeland@users.sourceforge.net>
 {% jdoc_nspace :jast java::lang.java.ast %}
 {% jdoc_nspace :jrule java::lang.java.rule %}
 
-{% include note.html content="TODO All that should be written in the Javadocs,
-not sure we even need a doc page. Would be simpler to maintain too" %}
-{% include warning.html content="WIP lots of stuff missing" %}
+{% include note.html content="Ideally most of what is written in this document would be directly
+in the Javadocs of the relevant classes. This is not the case yet." %}
+
+{% include warning.html content="There are still many topics missing, e.g. tree traversal using NodeStreams,
+using metrics, using type resolution, ..." %}
```

Comment:
node streams should be mentioned here, some pointers to javadoc are enough (`#descendants(Class)`, ...).

Using metrics/type res/symbols should point to java specific features, see #4319 / https://docs.pmd-code.org/latest/pmd_languages_java.html